fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/cursors/aggregate/AggregateAccumulators.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * AggregateAccumulators.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.cursors.aggregate;
+
+import com.apple.foundationdb.record.query.predicates.Value;
+
+import javax.annotation.Nonnull;
+
+/**
+ * A class that contains static initializers for various aggregate accumulators.
+ */
+public class AggregateAccumulators {
+
+    // ------------------ SUM Aggregators ------------------------------
+
+    public static RecordValueAccumulator<Integer, Integer> sumInteger(@Nonnull final Value value) {
+        return new RecordValueAccumulator<>(value, new IntegerState(PrimitiveAccumulatorOperation.SUM));
+    }
+
+    public static RecordValueAccumulator<Long, Long> sumLong(@Nonnull final Value value) {
+        return new RecordValueAccumulator<>(value, new LongState(PrimitiveAccumulatorOperation.SUM));
+    }
+
+    public static RecordValueAccumulator<Float, Float> sumFloat(@Nonnull final Value value) {
+        return new RecordValueAccumulator<>(value, new FloatState(PrimitiveAccumulatorOperation.SUM));
+    }
+
+    public static RecordValueAccumulator<Double, Double> sumDouble(@Nonnull final Value value) {
+        return new RecordValueAccumulator<>(value, new DoubleState(PrimitiveAccumulatorOperation.SUM));
+    }
+
+    // ------------------ MIN Aggregators ------------------------------
```

Comment:
MIN and MAX are meaningful for strings and byte arrays and so on, as well. Not needed in this initial cut, not at some point they ought to be added.

Reply:
Created rdar://83673890 (Streaming aggregation: MIN and MAX for Strings and Byte arrays)

Reply:
Created https://github.com/FoundationDB/fdb-record-layer/issues/1408